assignment/scripts/array-practice.js

```diff
@@ -75,20 +78,48 @@ console.log('The animals are now', animalArray);
 
 // 4.d TODO: Remove the food at the beginning of your array & 
 //     log both the food removed and the updated array
+removedFood = favoriteFoods.shift();
+console.log('Removed the first food', removedFood);
+console.log('The foods are now', favoriteFoods);
 
 // 4.e (STRETCH) TODO: Replace the second food in your array
 //      with another one of your favorite foods.
 //      Then log the updated array.
 
+/* found appropriate usage of .splice here:
+https://stackoverflow.com/a/586189
+=> modified to replace rather than insert
+ */
+favoriteFoods.splice(
+    1, // the index at which to splice
+    1, // number of elements to remove
+    'italian bread' // first element to insert
+)
+console.log('Replaced bread type', favoriteFoods);
+
 // 4.f (STRETCH) TODO: Sort your favoriteFoods array
 //     in reverse alphabetical order. Log the array.
+favoriteFoods.sort();
+console.log('Sorted foods:', favoriteFoods);
 
 // 4.g (STRETCH) TODO: Convert your array to a string
 //     putting the word "and" between each item.
 //     eg "tacos and pizza and pasta". Log the string.
+foodString = favoriteFoods.join(' and ');
+console.log('Food string:', foodString);
+/*
+Worked to figure out how to put commas in the right places.
+Imitated regex usage from here:
+https://stackoverflow.com/a/28673744
+No oxford comma, because I couldn't figure out a nice way of including it.
+*/
+foodString = foodString.replace(/ and(?=.* and)/g, ',');
+console.log('Food string:', foodString);
 
 // 4.h (STRETCH) TODO: Make a new array that combines 
 //     the favorite foods array with the animals array.
 //     Then log the new array.
 //     It should look something like:
-//     ['pizza', 'pasta', 'fish', 'cat', 'bird', 'dog']
\ No newline at end of file
+//     ['pizza', 'pasta', 'fish', 'cat', 'bird', 'dog']
+let exoticFoodArray = favoriteFoods.concat(animalArray);
```

Comment:
Nice. Could also have used the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) for this.